sdk/iot/Azure.Iot.Hub.Service/src/API Design.md

```diff
@@ -70,8 +70,53 @@ APIs for using IotHub v2 jobs
 </details>
 
 <details><summary><b>Messages</b></summary>
-Feedback messages, sending cloud to device messages (missing from current swagger), and purging cloud to device message queue
+Ssending cloud to device messages, receiving feedback messages, and purging cloud to device message queue
+
 ```csharp
+public class CloudToDeviceMessages
+{
+    /// <summary>
+    /// Send cloud to device message for a device.
+    /// </summary>
+    /// <param name="deviceId">The unique identifier of the device.</param>
+    /// <param name="message">The cloud to device message to be sent to the device. </param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>The Http response.</returns>
+    public virtual async Task<Response> SendMessageAsync(String deviceId, CloudToDeviceMessage message, CancellationToken cancellationToken = default) { }
+
+    /// <summary>
+    /// Retrieve feedback notification for cloud to device messages.
+    /// </summary>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>The message feedback batch, containing the details of the final state of sent messages.</returns>
```

Comment:
I don't know if this was brought up in the design meeting, but this call is more of a poll than a get. There is no guarantee that this method will return any feedback messages. This should be called out in the method documentation.

Reply:
> this call is more of a poll than a get

Could you explain a bit more about what you mean by this? 
Afaik, if the user sets the "ack" property in the c2d message, then the feedback message gets generated accordingly (I'll highlight this in the comments).
Service maintains a queue of feedback messages, which it returns to the user when a "get" call is made to this path "/messages/servicebound/feedback". There isn't a channel kept open to kept open where the sdk keeps looping to see if there is any new feedback received; so how is it polling?